modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/bindings/PolygonTypeBinding.java

```diff
@@ -98,15 +99,38 @@ public Class getType() {
     public Object parse(ElementInstance instance, Node node, Object value)
         throws Exception {
         //TODO: schema allows no exterior ring, but what the heck is that all about ?
-        LinearRing exterior = (LinearRing) node.getChildValue("exterior");
+        Object o = node.getChildValue("exterior");
+
+        LinearRing exterior = null;
+        if(o instanceof String){
```

Comment:
I'm confused, when does it happen that one gets back a String instead a geometry? By your tests it seems the ring case, but that looks to me more of an issue in another binding not working as it should.. like, a RingBinding should be returning a LinearRing, the bug imho should be searched in there and/or in the CurvedGeometryFactory.